assets/src/blocks/SocialMedia/SocialMediaEditorScript.js

```diff
@@ -0,0 +1,244 @@
+  return new Promise((resolve, reject) => {
+    let tag = document.createElement('script');
+    tag.src = uri;
+    tag.async = true;
+    tag.onload = () => {
+      resolve();
+    };
+    const body = document.getElementsByTagName('body')[0];
+    body.appendChild(tag);
+  });
+};
+
+const initializeTwitterEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.twttr) {
+      window.twttr.widgets.load();
+    }
+  }, 2000);
+}
+
+const initializeInstagramEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.instgrm) {
+      window.instgrm.Embeds.process();
+    }
+  }, 3000);
+}
+
+const initializeFacebookEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.FB) {
+      window.FB.XFBML.parse();
+    }
+  }, 3000);
+}
+
+const PROVIDER_SCRIPT_DATA = {
+  twitter: {
+    script: 'https://platform.twitter.com/widgets.js',
+    initFunction: initializeTwitterEmbeds,
+  },
+  facebook: {
+    script: 'https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v9.0',
+    initFunction: initializeFacebookEmbeds,
+  },
+  instagram: {
+    script: 'https://www.instagram.com/embed.js',
+    initFunction: initializeInstagramEmbeds,
+  },
+};
+
+export const SocialMediaEditor = ({
+  attributes,
+  isSelected,
+  setAttributes,
+}) => {
+  const {
+    social_media_url,
+    facebook_page_tab,
+    description,
+    title,
+    embed_type,
+    alignment_class,
+    embed_code,
+    className,
+  } = attributes;
+
+  const toAttribute = attributeName => value => setAttributes({
+    [attributeName]: value
+  });
+
+  /**
+   * Check if social media corresponding embeds script is loaded and initiliaze it.
+   * Can be used for Facebook, Twitter and Instagram depending on the parameter.
+   */
+  const checkProviderScript = provider => {
+    const providerData = PROVIDER_SCRIPT_DATA[provider];
+    const script = document.querySelector(`body > script[src="${providerData.script}"]`);
+    if (script === null) {
+      const scriptLoaded = loadScriptAsync(providerData.script);
+      scriptLoaded.then(providerData.initFunction);
+    } else {
+      providerData.initFunction();
+    }
+  }
+
+  const updateEmbed = async (url, provider) => {
+    if (!url || provider === 'facebook') {
+      setAttributes({ embed_code: '' });
+      return;
+    }
+    let embedCode;
+    try {
+      if (provider === 'twitter') {
+        const twitterEmbedData = await apiFetch({ path: addQueryArgs('/oembed/1.0/proxy', { url }) });
+        embedCode = twitterEmbedData ? twitterEmbedData.html : '';
+      } else if (provider === 'instagram') {
+        const instagramEmbedData = await fetchJson(`${INSTAGRAM_OEMBED}?url=${url}&access_token=${FB_ACCESS_TOKEN}`);
+        embedCode = instagramEmbedData ? instagramEmbedData.html : '';
+      }
+    } catch (error) {
+      embedCode = '';
+    }
+    setAttributes({ embed_code: embedCode });
+  };
+
+  useEffect(() => {
+    ALLOWED_OEMBED_PROVIDERS.forEach(provider => {
```

Comment:
This would be easier to follow if not everything happened within the `forEach`. You could split it like this:
```js
const provider = ALLOWED_OEMBED_PROVIDERS.find(provider => social_media_url.includes(provider);
if (!provider) {
  setAttributes({ embed_code: '' });
  return;
}
// ... other stuff

Reply:
Updated!